### ISSUE TITLE
Second Screen WG/CG - TPAC 2021 agenda

### ISSUE THREAD

**Issue by anssiko** [w3c/secondscreen-wg#3]
>🚧 This issue is to work out an agenda for the [Second Screen WG/CG](https://www.w3.org/2014/secondscreen/) [TPAC 2021](https://www.w3.org/2021/10/TPAC/) virtual meeting.

# Registration

TPAC registration will open on 20 September 2021, register via https://www.w3.org/2021/10/TPAC/

# Logistics

CONFIRMED dates and times, 1-2 hours a day, subject to the agenda:
- ⏰ 25 Oct 2021 14:00-15:00+ UTC (see [your local time](https://www.timeanddate.com/worldclock/fixedtime.html?iso=20211025T14)) - [🗓️ Calendar](https://www.w3.org/events/meetings/cbb60781-e18b-4fd9-b6bf-464df6b6ad03)
- ⏰ 26 Oct 2021 05:00-06:00 UTC (see [your local time](https://www.timeanddate.com/worldclock/fixedtime.html?iso=20211026T05)) - [🗓️ Calendar](https://www.w3.org/events/meetings/b2d24686-103a-4eb9-8499-b3a35a4771a7)
- 📞 For Zoom details, see the meeting Calendar
- ✍️ [IRC](https://irc.w3.org/?channels=#webscreens) or irc://irc.w3.org:6667/#webscreens

# Agenda (🚧 WIP)

> Feedback and suggestion on specific agenda items welcome via comments. 👇

## 25 Oct 2021 14:00-15:00+ UTC - Second Screen WG

| ⏰ UTC time |  |
| :---: | :--- |
| 14:00 UTC | Welcome, agenda bashing |
| UTC | Open Screen Protocol: [proposed changes](https://github.com/w3c/openscreenprotocol/pulls) since FPWD, [security-tracker feedback](https://github.com/w3c/openscreenprotocol/issues?q=is%3Aissue+is%3Aopen+label%3Asecurity-tracker), [wide review](https://github.com/w3c/openscreenprotocol/issues/284) (Mark Foltz @mfoltzgoogle) |
| UTC | Presentation API: https://github.com/w3c/presentation-api/issues/485 (Francois Daoust @tidoust)  |
| UTC | Remote Playback API: Proposed Rec https://github.com/w3c/remote-playback/issues/130 blocked on test automation https://github.com/w3c/remote-playback/issues/92 (Louay Bassbouss @louaybassbouss) |
| UTC | WG Charter 2022: prep for charter update, review CG incabations for WG readiness, discuss OSP changes subject to the PAG progress, see [charter 2019-2021](https://www.w3.org/2020/12/second-screen-wg-charter.html) (chairs @anssiko @louaybassbouss W3T @tidoust) |
| 15:00 UTC | [Multicast CG](https://www.w3.org/community/multicast/) Intro & discussion (Jake Holland @GrumpyOldTroll) |

## 26 Oct 2021 05:00-06:00 UTC - Second Screen CG

| ⏰ UTC time |  |
| :---: | :--- |
| 14:00 UTC | Multi-Screen Window Placement: key [open issues](https://github.com/webscreens/window-placement/issues/), [Origin Trial feedback](https://groups.google.com/a/chromium.org/g/blink-dev/c/jznxQK1U8ZQ/m/fOjP70c7AQAJ) / [developer feedback](https://github.com/webscreens/window-placement/issues/67) dissemination (Michael Wasserman @michaelwasserman) |
| UTC | Window Segments Enumeration: [deprecated](https://github.com/webscreens/window-segments/pull/15) by the new [Viewport Segments Property](https://github.com/WICG/visual-viewport/blob/gh-pages/segments-explainer/SEGMENTS-EXPLAINER.md) proposal, intro and discussion (Daniel Libby @dlibby- Stephanie Stimac @ststimac? ) |
| UTC | Presentation API [v2 features](https://github.com/w3c/presentation-api/issues?q=is%3Aissue+is%3Aopen+label%3Av2) and future work: [Tab Self Mirroring API](https://mfoltzgoogle.github.io/tab-self-mirroring/explainer.html) (Mark Foltz @mfoltzgoogle) |
| UTC | Remote Playback API v2 features: none |

## 11 January 2022 22:00 UTC / 14:00 US Pacific - Joint meeting with Media WG

| ⏰ UTC time |  |
| :---: | :--- |
| 22:00 UTC | https://github.com/w3c/media-wg/issues/33 |

# Minutes

- [Second Screen TPAC 2021 virtual - Day 1/2 minutes](https://www.w3.org/2021/10/25-webscreens-minutes.html)
- [Second Screen TPAC 2021 virtual - Day 2/2 minutes](https://www.w3.org/2021/10/26-webscreens-minutes.html)

# Participants

All Second Screen WG and CG participants welcome.

**Comment by markafoltz** [comment]
Hi @anssiko thanks for kicking off the agenda.  Here is some feedback, I hope this helps fill out some of the details.

> Open Screen Protocol: proposed changes since FPWD, security-tracker feedback, wide review

I would like to discuss PR w3c/openscreenprotocol#286 from Apple.  I also plan prepare a short deck responding to the (many) security-tracker feedback item.  Those are likely the highest priority items to get group input on from my perspective.

For the wide review, I can start some PRs responding to the checklists, and we can use the remaining time to get feedback on them from the group.

My best guess is this will take ~30 minutes

>  Presentation API

I don't have any specific issues to bring forward.  I think w3c/presentation-api#485 is ready for a PR, and @tidoust mentioned he would give it a try, so if there is one to review happy to look at it.   ~10 minutes if we need to discuss a PR

> Remote Playback API

The main open item I see is the meta-issue for advancement to PR (w3c/remote-playback#130).   The last comment suggested @louaybassbouss was going to check on test automation for this API (w3c/remote-playback#92).   I suggest we spend some time on that as well as any other items on the PR checklist that need group input.    ~20 minutes

> Tab Self-Mirroring

We'll provide more details on this item in an update next week.

> Remote Playback API v2

I don't have anything for the agenda.

**Comment by anssiko** [comment]
Thanks @mfoltzgoogle, the agenda is now updated accordingly.

@mfoltzgoogle @tidoust @louaybassbouss @GrumpyOldTroll @michaelwasserman @dlibby-:
Please check your availability for your topics (https://github.com/w3c/secondscreen-wg/issues/3#issue-972534039) and chime in here if you'd like us to adjust.

I'm happy to share we have a guest speaker, Jake Holland, [Multicast CG](https://www.w3.org/community/multicast/) chair, on Day 1 to talk about Multicast CG working to enable multicast IP transport for web traffic. In particular, Jake was interested in discussing with us the following given we have a shared problem space of adding new protocols to a web browser:

- privacy issues and mitigations for the between-device discovery on the LAN
- learnings from exposing site-controlled information on the LAN

For context, there's a thread over at [Chromium net-dev](https://groups.google.com/a/chromium.org/g/net-dev/c/TjbMyPKuRHs/m/_Syfhri7AAAJ).

@grumpyoldtroll, feel free to use this issue to document any other questions you'd like the Second Screen folks to think before our meeting. Thanks for joining us!

All - We're also planning another joint meeting with Media WG to resolve https://github.com/w3c/media-wg/issues/33 but moved that outside the TPAC week given scheduling challenges. We'll inform you of that separately.

**Comment by louaybassbouss** [comment]
Thanks @anssiko agenda LGTM

**Comment by GrumpyOldTroll** [comment]
Thanks @anssiko , LGTM.  The slot is in my calendar.

The privacy issue you cited is the one I most want to pick this group's brain about.  I'll also give a very brief intro to the [multicast cg](https://github.com/w3c/multicast-cg), which interested people are invited to join. And of course I'll be happy to answer any questions.

One probable FAQ (I imagine) would be about jointly subscribing multiple screens to multicast to reduce local network load for a 2nd screen use case.

That might be helpful in wired or cellular contexts and for wifi contexts with enough subscribers, but for anyone who wants to dig on that question, I would urge you to give a read to what is [imminently](https://www.rfc-editor.org/auth48/rfc9119) to be published as [RFC 9119](https://www.rfc-editor.org/rfc/rfc9119.html) (but at the time of this comment still needs to be viewed in its [draft form](https://www.ietf.org/archive/id/draft-ietf-mboned-ieee802-mcast-problems-15.html))

In short: it probably needs to avoid doing so for devices connected to a LAN with wi-fi unless there are enough devices, and it'll be moderately complicated up thru wi-fi 6 at least. So to anyone thinking about that use case: please take a look at the ietf doc if you want an intelligible discussion.  Just wanted to head off that rathole if possible, as it suggests the easy and obvious overlap between our groups might be lower than we'd wish for.

**Comment by anssiko** [comment]
Thanks @GrumpyOldTroll, we'll start with Multicast as soon as we've covered the WG business. We'll block an extra hour (25 Oct 2021, 15-16 UTC) on top of the WG agenda for Multicast, but may be also able to start sooner and finish sooner.

Invites should arrive shortly with Zoom coordinates.

**Comment by dlibby-** [comment]
Just to clarify, is the second day 14:00-15:00 UTC as listed in the table, or 05:00-06:00 UTC as listed under the Logistics bullets? 14:00 UTC is pretty rough for me (I'm in HST, so 4am local time). 05:00 UTC is more manageable (7pm local time). I can also ask @ststimac if she'd like to attend as she's been driving much of the standardization efforts recently.

**Comment by anssiko** [comment]
@dlibby- 26 Oct 2021 05:00-06:00 UTC is the correct time. Now fixed and updated invitation sent.

Thanks for catching the bug!

**Comment by chrisn** [comment]
I have to send apologies for the first meeting, but I will join the second meeting. I'm mostly interested in the Open Screen Protocol discussion but it seems will have to miss that.
